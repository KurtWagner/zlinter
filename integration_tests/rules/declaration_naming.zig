pub var BadName = 10;
pub var BadNameAgain: u32 = undefined;
pub var GoodOne: ?type = null;
pub var GoodOneAgain: ?type = null;

pub var MyStruct = struct {
    good_name: u32,
    BadName: u32,
    OkName: type,
    AnotherOkName: ?type,
    YetAnotherOkName: ?type = null,
    badName: []u8,
    AnotherBadOne: type = undefined,
    gameDraw: *const fn () void = undefined,
};

pub const MyGoodEnum = struct { a, a_b };
pub const MyBadEnum = struct { A };
pub const MyBadEnumAgain = struct { aB };

pub var gameDraw: *const fn () void = undefined;
pub var badNum: u32;
pub var anotherBadNum: u32 = 10;
pub var YetAnotherBadNum: usize = 10;
