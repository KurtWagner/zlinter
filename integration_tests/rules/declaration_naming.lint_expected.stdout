error Variable declaration should be snake_case [rules/declaration_naming.input.zig:1:9] declaration_naming

 1 | pub var BadName = 10;
   |         ^^^^^^^

error Variable declaration should be snake_case [rules/declaration_naming.input.zig:2:9] declaration_naming

 2 | pub var BadNameAgain: u32 = undefined;
   |         ^^^^^^^^^^^^

error Variable declaration should be snake_case [rules/declaration_naming.input.zig:3:9] declaration_naming

 3 | pub var GoodOne: ?type = null;
   |         ^^^^^^^

error Variable declaration should be snake_case [rules/declaration_naming.input.zig:4:9] declaration_naming

 4 | pub var GoodOneAgain: ?type = null;
   |         ^^^^^^^^^^^^

error Variable declaration should be snake_case [rules/declaration_naming.input.zig:22:9] declaration_naming

 22 | pub var badNum: u32;
    |         ^^^^^^

error Variable declaration should be snake_case [rules/declaration_naming.input.zig:23:9] declaration_naming

 23 | pub var anotherBadNum: u32 = 10;
    |         ^^^^^^^^^^^^^

error Variable declaration should be snake_case [rules/declaration_naming.input.zig:24:9] declaration_naming

 24 | pub var YetAnotherBadNum: usize = 10;
    |         ^^^^^^^^^^^^^^^^

error Constant declaration should be snake_case [rules/declaration_naming.input.zig:26:7] declaration_naming

 26 | const shouldBeTitleCase: ?type = null;
    |       ^^^^^^^^^^^^^^^^^

x 8 errors
